### PR TITLE
Add deck editor

### PR DESCRIPTION
**Pull request by derricklee43**
It is still kind of a mess🥲
I have logged most of the issues in trello, and left todo's, but I plan on going over it a lot more thoroughly after the demo and adding testing. 

https://user-images.githubusercontent.com/39753553/160962009-c360d478-8c94-4514-a580-fca2a2426fd2.mov

### CODE REVIEW

File: src/app.scss [hunk 1]
```diff
@@ -6,14 +6,19 @@
 }
 
 .App {
+  .text-areas {
```
Comment (jiejasonliu) [inline src/app.scss:9]: This is duplicated, probably some auto-merge trickery here -- probably safe to leave this here if we want to merge asap tho

File: src/components/deck-editor/meta-data-editor/meta-data-editor.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+import { Deck } from '../../../models/deck';
+import { Button } from '../../button/button';
+import { TextBox } from '../../text-box/text-box';
+import { BubbleDivider } from '../../bubble-divider/bubble-divider';
+import { DropDown, DropDownOption } from '../../drop-down/drop-down';
+import { Language } from '../../../models/card-content';
+import { TextArea } from '../../text-area/text-area';
+
+// Todo: maybe change Language to be enum instead of type
+enum Languages {
```
Comment (jiejasonliu) [inline src/components/deck-editor/meta-data-editor/meta-data-editor.tsx:12]: For future reference, we have this as a type in `card-content.ts`

File: src/components/drop-down/drop-down-choice.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+interface DropDownChoiceProps {
+  variant?: 'dark' | 'light';
+  className?: string;
+  bubbleOnClickEvent?: boolean;
+  onClick: (event: React.MouseEvent) => void;
+}
+
+export const DropDownChoice = ({
```
Comment (jiejasonliu) [inline src/components/drop-down/drop-down-choice.tsx:9]: We don't need this right?

**Review by jiejasonliu — APPROVED** [review]
LGTM, if it's not too much work you can address some comments, otherwise you're free to merge since we're gonna be fixing a lot of this up anyway

File: src/components/deck-editor/meta-data-editor/meta-data-editor.tsx (hunk 2 above)
Comment (derricklee43) [reply src/components/deck-editor/meta-data-editor/meta-data-editor.tsx:12]: Yeah, that's what I meant, haha.

But if we need to keep it is a type, we could create a Record<Language, string> object in the card-content.ts file and update them together

File: src/app.scss (hunk 1 above)
Comment (derricklee43) [reply src/app.scss]: haha, whoops. yep, I got it removed

File: src/components/deck-editor/meta-data-editor/meta-data-editor.tsx (hunk 2 above)
Comment (jiejasonliu) [reply src/components/deck-editor/meta-data-editor/meta-data-editor.tsx:12]: I think at that point an enum is probably better

**Pull request merged**